feat(App.tsx): add feature edit

ismalzikri opened the pull request:
there are some additional and several changes :

- additional edit feature with input field
- icon edit 

out of the title in this commit : 

- clear all items

ismalzikri committed:
feat(App.tsx): add feature edit

additional edit features

mhaidarhanif commented:
Clickable area to edit can be better for the whole row of todo item. So it's not as hard as needing to click the text only. Especially if the todo text is so short.

mhaidarhanif commented:
Also, find a way to keep the height of the todo item card to be consistent when exist (not being edited) and when being edited.

Currently the editing state cause the height to be bigger.

ismalzikri committed:
perf(App.tsx, App.css): update clickable area item todo

ismalzikri committed:
refactor(App.tsx): avoid twice clicked to show the input field

after this changes, user only clicked for a once to show the input field
with item inside of it
